cupyx/scipy/ndimage/interpolation.py

```diff
@@ -386,15 +365,37 @@ def shift(input, shift, output=None, order=None, mode='constant', cval=0.0,
 
     _check_parameter('shift', order, mode)
 
-    if mode == 'opencv':
-        mode = '_opencv_edge'
-
     if not hasattr(shift, '__iter__') and type(shift) is not cupy.ndarray:
         shift = [shift] * input.ndim
 
-    return affine_transform(input, cupy.ones(input.ndim, input.dtype),
-                            cupy.negative(cupy.asarray(shift)), None, output,
-                            order, mode, cval, prefilter)
+    if mode == 'opencv':
+        mode = '_opencv_edge'
+
+        output = affine_transform(
+            input,
+            cupy.ones(input.ndim, input.dtype),
+            cupy.negative(cupy.asarray(shift)),
+            None,
+            output,
+            order,
+            mode,
+            cval,
+            prefilter,
+        )
+    else:
+        if order is None:
+            order = 1
+        output = _get_output(output, input)
+        if input.dtype.kind in "iu":
```

Comment:
Could you use single quotes for our convention?
```suggestion
        if input.dtype.kind in 'iu':
```